src/components/TodoApp.tsx

```diff
@@ -0,0 +1,221 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+
+import React, {
+  useContext, useEffect, useMemo, useRef, useState,
+} from 'react';
+import cn from 'classnames';
+
+import { TodoList } from './TodoList';
+import { DispatchContext, StateContext } from './TodosProvider';
+import { Footer } from './Footer';
+import { ErrorMessage } from '../types/ErrorMessage';
+import { USER_ID } from '../constants/userId';
+import { createTodo, updateTodo } from '../api/todos';
+import { TodoItem } from './TodoItem';
+import { LoadingStatus } from '../types/LoadingStatus';
+import { TEMP_TODO_ID } from '../constants/tempTodoId';
+
+export const TodoApp = () => {
+  const {
+    todos,
+    errorMessage,
+    tempTodo,
+  } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const completedTodosCount = useMemo(() => (
+    todos.filter(
+      todo => todo.completed,
+    ).length
+  ), [todos]);
+
```

Comment:
Try do not repeat your self. Is possible to pas this value usign pros? Footer component has same logic